### ISSUE TITLE
TS rules should only apply to .ts and .vue files with <script lang="ts"> in mixed codebase

### ISSUE THREAD

**Issue by dsteinbach** [typescript-eslint/typescript-eslint#1835]
**Repro**

```JS
module.exports = {
    root: true,
    env: {
        node: true,
    },
    extends: ['eslint:recommended', 'plugin:vue/recommended', '@vue/typescript'],
    rules: {
        '@typescript-eslint/explicit-function-return-type': 'off',
    },
    parserOptions: {
        parser: '@typescript-eslint/parser'
    },
    overrides: [
        {
            files: ['*.ts'],
            rules: {
                '@typescript-eslint/explicit-function-return-type': ['error'],
            },
        },
    ],
};

```

```
<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
    ...
    methods: {
        // undefined return type
        someFunc() {
             return 'abc';
        }
    },
    watch: {
        clientCode(val: string): here-is-an-undefined-return-type {
              this.someStateVar = val;     
        },
    },
});
</script>

```
**Additional Info**
If I get rid of overrides object and enable `explicit-function-return-type` under the default rules object that rule will run on all .vue files regardless if it's `<script lang="ts">` or `<script lang="js">`.  Everything works as it should if the file extension is .ts.

**Expected Result**
Should show error for .vue files that have `<script lang="ts">` and no return type explicitly defined.

**Actual Result**
Does not show error for .vue files that have `<script lang="ts">` with no return type explicitly defined.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^2.19.0` |
| `TypeScript`                | `^3.7.3` |
| `ESLint`                    | `^6.7.2` |
| `node`                      | `8.16.2` |
| `npm`                       | `6.4.1` |

**Comment by bradzacher** [comment]
A .vue file isn't going to match a .ts extension glob because the extensions don't match.

**Comment by dsteinbach** [comment]
If i add vue to my glob then it will run TS rules on .vue files that contain both `<script lang="ts">` and `<script>` (js).

**Comment by bradzacher** [comment]
you should file a request in the vue parser. our parser has no concept of what a vue file even is. the vue parser is what converts your vue code to something consumable by our parser.
